From ee655791d3beec31be6b311fce19e9e7a9de4b60 Mon Sep 17 00:00:00 2001
From: Nachiket Sahasrabuddhe <nachiket_sahasrabuddhe@xyratex.com>
Date: Wed, 20 Feb 2013 12:57:12 +0530
Subject: [PATCH 4/6] cookie-replacing-senderID

1. Moved m0_xc_cookie_init() from rpc/session_fops.c to lib/m0lib.c
---
 lib/m0lib.c        | 3 +++
 rpc/session_fops.c | 1 -
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/m0lib.c b/lib/m0lib.c
index f44bd2c..834afde 100644
--- a/lib/m0lib.c
+++ b/lib/m0lib.c
@@ -21,17 +21,20 @@
 #include "lib/cdefs.h"
 #include "lib/buf_xc.h"
 #include "lib/vec_xc.h"
+#include "lib/cookie_xc.h"
 
 static void xc_types_init(void)
 {
 	m0_xc_buf_init();
 	m0_xc_vec_init();
+	m0_xc_cookie_init();
 }
 
 static void xc_types_fini(void)
 {
 	m0_xc_buf_fini();
 	m0_xc_vec_fini();
+	m0_xc_cookie_fini();
 }
 
 M0_INTERNAL int libm0_init(void)
diff --git a/rpc/session_fops.c b/rpc/session_fops.c
index edd3079..cce498c 100644
--- a/rpc/session_fops.c
+++ b/rpc/session_fops.c
@@ -146,7 +146,6 @@ M0_INTERNAL int m0_rpc_session_fop_init(void)
 	 * @todo This should be done from dtm subsystem init.
 	 */
 	m0_xc_verno_init();
-	m0_xc_cookie_init();
 	m0_xc_rpc_onwire_init();
 	m0_xc_session_fops_init();
 	return  M0_FOP_TYPE_INIT(&m0_rpc_fop_conn_establish_fopt,
-- 
1.8.3.2

